agenta-cli/agenta/sdk/config_manager.py

```diff
@@ -0,0 +1,205 @@
+import json
+import logging
+from pathlib import Path
+from typing import Optional, Type, TypeVar
+
+import yaml
+from pydantic import BaseModel, ValidationError
+
+from agenta.client.backend.client import AgentaApi
+from agenta.sdk.decorators.llm_entrypoint import route_context
+
+from . import AgentaSingleton
+
+T = TypeVar("T", bound=BaseModel)
+
+logger = logging.getLogger(__name__)
+singleton = AgentaSingleton()
+
+
+class ConfigManager:
+    client = None
```

Comment:
This `client` is a class variable.
Given that `client` uses an `api-key` and eventually a `project-id`, it is effectively a singleton tied to a specific project.
If the `project-id` and the `api-key` will be provided in `ag.init`, why not create a singleton object like `ag.tracing`, so;ething like `ag.config(s)`, that only stores the client in its state ?

Reply:
The `AgentaSingleton`, yes, but not an `ConfigManager` singleton ?

Reply:
Yes, I am reusing the same AgentaSingleton